MenuTest/ViewController.swift

```diff
@@ -14,7 +14,7 @@ class ViewController: UIViewController {
     override func viewDidLoad() {
         super.viewDidLoad()
         
-        let menu = MenuView(title: "Menu", theme: LightMenuTheme()) { () -> [MenuItem] in
+        let menu = MenuView(title: "Menu", theme: LightMenuTheme()) { [weak self] () -> [MenuItem] in
```

Comment:
The [weak self] here is unnecessary because self is unused in the containing closure. Better to be explicit about when to use it then to use it everywhere out of an abundance of caution

Reply:
That’s what I initially thought, but the memory graph in Xcode shows MenuView captures the closure capture ViewController (I tested this by having another view controller present the ViewController)

And yes, unnecessary self doesn’t capture self.

Reply:
Oh wow you're totally right! Why does this happen? Is the outer closure capturing `self` strongly in order for it to be used in the inner closure? 

I'll merge this, thank you for the fix.

Reply:
I was also wondering, and just found an article online: 
https://blog.haloneuro.com/swift-memory-leak-gotcha-with-weak-self-67293d5bc060

Reply:
Now I have to go through a ton of other code and check for that type of leak. Thanks 😛